File: content/examples/other/leetcode/problems/sum-of-two-integers/example-2/solution.py
def get_sum(a, b):
    mask = 0xFFFFFFFF
    a &= mask
    b &= mask
    while b != 0:
        carry = ((a & b) << 1) & mask
        a = (a ^ b) & mask
        b = carry
    if a > 0x7FFFFFFF:
        a = ~ (a ^ mask)
    return a


if __name__ == "__main__":
    print(get_sum(1, 2))
    print(get_sum(-4, 7))

File: content/examples/other/leetcode/problems/sum-of-two-integers/example-2/solution.js
function getSum(a, b) {
  a |= 0;
  b |= 0;
  while (b !== 0) {
    const carry = (a & b) << 1;
    a = (a ^ b) | 0;
    b = carry | 0;
  }
  return a | 0;
}

if (require.main === module) {
  console.log(getSum(1, 2));
  console.log(getSum(-4, 7));
}

module.exports = { getSum };
